src/decoder.rs

```diff
@@ -56,6 +54,16 @@ where
         }
     }
 
+    /// Unwraps the inner read source.
+    pub fn unwrap(self) -> R {
```

Comment:
The idiomatic name for something like this is usually `into_inner`. e.g., https://doc.rust-lang.org/std/?search=into_inner

Reply:
Thanks, done.